master.js

```diff
@@ -1007,15 +962,74 @@ async function getPlugins(){
 		let plugin = plugins[i];
 		if(plugin.main.onLoadFinish && typeof plugin.main.onLoadFinish == "function") await plugin.main.onLoadFinish({plugins});
 	}
-	console.log("All plugins loaded in "+(Date.now() - startPluginLoad)+"ms");
 	return plugins;
 }
 
+async function startServer() {
+	// Set the process title, shows up as the title of the CMD window on windows
+	// and as the process name in ps/top on linux.
+	process.title = "clusterioMaster";
+
+	// add better stack traces on promise rejection
+	process.on('unhandledRejection', r => console.log(r));
 
-masterPlugins = await getPlugins();
+	/** Sync */
+	function randomStringAsBase64Url(size) {
+	  return base64url(crypto.randomBytes(size));
+	}
+
+	if (!fs.existsSync("secret-api-token.txt")) {
+		config.masterAuthSecret = randomStringAsBase64Url(256);
+		fs.writeFileSync("config.json",JSON.stringify(config, null, 4));
+		fs.writeFileSync("secret-api-token.txt", jwt.sign({ id: "api" }, config.masterAuthSecret, {
+			expiresIn: 86400*365 // expires in 1 year
+		}));
+		console.log("Generated new master authentication private key!");
+		process.exit(0);
+	}
+	// write an auth token to file
+	fs.writeFileSync("secret-api-token.txt", jwt.sign({ id: "api" }, config.masterAuthSecret, {
+		expiresIn: 86400*365 // expires in 1 year
+	}));
+	setInterval(()=>{
+		fs.writeFileSync("database/items.json", JSON.stringify(db.items));
+	},config.autosaveInterval || 60000);
+	process.on('SIGINT', shutdown); // ctrl + c
+	process.on('SIGHUP', shutdown); // terminal closed
+
+	// Make sure we're actually going to listen on a port
+	let httpPort = args.masterPort || config.masterPort;
+	let httpsPort = args.sslPort || config.sslPort;
+	if (!httpPort && !httpsPort) {
+		console.log("Error: no port configured to listen to");
+		process.exit(1);
```

Comment:
This is the convention in 4 other places.  Setting exit code and waiting for it to die doesn't work as there's a minefield of interval and timed callbacks everywhere, so node doesn't exit.

Reply:
@Hornwitser well, the thing is that we probably should rewrite stuff to do proper exiting have timeouts or/and intervals exit properly. Existing like this can lose data or get into an inconsistent state.
So I do agree for this PR this is okay, but we probably should have an issue for a refactoring.